Android/app/src/main/java/com/peer1/internetmap/models/ASN.java

```diff
@@ -6,10 +6,21 @@
  * Created by shayla on 2017-05-10.
  */
 public class ASN {
+
+    private final String none = "none";
+
     @SerializedName("resultsPayload")
-    public Integer asn;
+    public String asn;
 
+    /**
+     * Note API returns "none" in cases where no asn can be found.
+     * @return null if no asn found, else the asn found
+     */
     public String getASNString() {
+        if (asn.isEmpty() || asn.equals(none)) {
+            return null;
+        }
+
         return String.valueOf(asn);
```

Comment:
Looking into valueOf it appears to return the string "null" if the String is null; this was probably causing me UI troubles since I added the code to return null (not a string). 

As such, you're probably right, we should be able to simply return asn here.